scripts/create-darwin-volume.sh

```diff
@@ -127,42 +164,75 @@ main() {
+    if ! test_fstab; then
+        echo "Configuring /etc/fstab..." >&2
+        label=$(echo "$volume" | sed 's/ /\\040/g')
+        # shellcheck disable=SC2209
+        printf "\$a\nLABEL=%s /nix apfs rw,nobrowse\n.\nwq\n" "$label" | EDITOR=ed sudo vifs
+    fi
 
-        if test_filevault_in_use; then
-            # TODO: Not sure if it's in-scope now, but `diskutil apfs list`
-            # shows both filevault and encrypted at rest status, and it
-            # may be the more semantic way to test for this? It'll show
-            # `FileVault:                 No (Encrypted at rest)`
-            # `FileVault:                 No`
-            # `FileVault:                 Yes (Unlocked)`
-            # and so on.
-            if test_t2_chip_present; then
-                echo "warning: boot volume is FileVault-encrypted, but the Nix store volume" >&2
-                echo "         is only encrypted at rest." >&2
-                echo "         See https://nixos.org/nix/manual/#sect-macos-installation" >&2
-            else
-                echo "error: refusing to create Nix store volume because the boot volume is" >&2
-                echo "       FileVault encrypted, but encryption-at-rest is not available." >&2
-                echo "       Manually create a volume for the store and re-run this script." >&2
-                echo "       See https://nixos.org/nix/manual/#sect-macos-installation" >&2
-                exit 1
-            fi
+    if [ -n "$create_volume" ]; then
+        echo "Creating a Nix volume..." >&2
+
+        sudo diskutil apfs addVolume "$disk" "$NIX_VOLUME_FS" "$volume" -mountpoint /nix
+        new_uuid="$(volume_uuid "$volume")"
+
+        if [ "$INSTALL_MODE" = "no-daemon" ]; then # exported by caller
+            # TODO: is there a better way to do this?
+            sudo chown $USER:admin /nix
         fi
 
-        sudo diskutil apfs addVolume "$disk" APFS 'Nix Store' -mountpoint /nix
-        volume="Nix Store"
+        if test_filevault_in_use; then
+            # security program's flags won't let us both specify a keychain
+            # and be prompted for a pw to add; two step workaround:
+            # 1. add a blank pw to system keychain
+
+            # system is in some sense less secure than user keychain... (it's
+            # possible to read the password for decrypting the keychain) but
+            # the user keychain appears to be available too late. As far as I
+            # can tell, the file with this password (/var/db/SystemKey) is
+            # inside the FileVault envelope. If that isn't true, it may make
+            # sense to store the password inside the envelope?
+            sudo /usr/bin/security add-generic-password -a "$volume" -s "$new_uuid" -D "$volume encryption password" -j "Added automatically by the Nix installer for use by /Library/LaunchDaemons/org.nixos.darwin-store.plist" "/Library/Keychains/System.keychain"
```

Comment:
The label will default to the service name. We should provide an explicit label instead.

And the kind should probably be something generic instead of including the volume name.

Here's what I'd recommend:

```suggestion
            sudo /usr/bin/security add-generic-password -a "$volume" -s "$new_uuid" -l "$volume encryption password" -D "Encrypted volume password" -j "Added automatically by the Nix installer for use by /Library/LaunchDaemons/org.nixos.darwin-store.plist" "/Library/Keychains/System.keychain"
```

Reply:
I'll tentatively try this out and generate a new installer from it, but I'm leaving the comment open for now. 

The label was giving me a lot of grief at some point in development (I think it had something to do with what fields security was willing to recognize for the -U/update op?), but I would prefer a distinct label to fiddling with the kind, and I already have to hoop-jump to test other changes, so it's cheap to give it a roll.